File: Lista nr 3/Pentagon.cpp
#include "Pentagon.hpp"
#define Constant 1.72048
#include "NegativeNumber.hpp"
Pentagon::Pentagon(int a)
{
    if(a < 0)
        throw NegativeNumber();
    side = a;
}

int Pentagon::area()
{
    return side * side * Constant;
}
int Pentagon::perimeter()
{
    return 5 * side;
}
std::string Pentagon::getNAME()
{
    return "Pentagon";
}

Pentagon::~Pentagon()
{

}

File: Lista nr 3 Java/Pentagon.java
class Pentagon implements Figure
{
    private static final double Constant = 1.72048;
    private int side;
    public Pentagon(int a) throws NegativeNumber
    {
        if(a < 0)
        throw new NegativeNumber();
        side = a;
    }
    public int area()
    {
        return (int)(side * side * Constant);
    }
    public int perimeter()
    {
        return 5 * side;
    }
    public String getNAME()
    {
        return "Pentagon";
    }
}